doc/custom_nonconformity_score_proof.ipynb

```diff
@@ -0,0 +1,407 @@
+   "cell_type": "markdown",
+   "id": "a3995772",
+   "metadata": {},
+   "source": [
+    "The purpose of this documentation is to generalize the coverage guarantees of the [Predictive inference with the jackknife+](https://arxiv.org/pdf/1905.02928.pdf) methods (jacknife+ and jacknife mimax) to a larger set of nonconformity scores.\n",
+    "\n",
+    "First, two function sets are definied with properties that allow to verify the same coverage properties. Then, the two theorems guaranteing coverage for the jackknife+ method and the jackknife minmax one are reviewed with these new sets of functions."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1615ad39",
+   "metadata": {},
+   "source": [
+    "___\n",
+    "# Nonconformity score and predition function\n",
+    "In order to generalize the coverage guarantee to more nonconformity scores (not only the absolute residuals), let first define two functions with some properties. The nonconformity score $f$ defined on $E^{2}$ ($E \\subset \\mathbb{R}$) and the prediction function $g$ defined on $F \\times E$ ($F \\subset \\mathbb{R}$). Uppercase letters are used to stick to the paper notation.\n",
+    "\n",
+    "## Nonconformity score\n",
+    "From an observation and a prediction, the function $f$ outputs the conformity score:\n",
+    "$$\\begin{align}\n",
+    "f:E^{2} &\\to F\\\\\n",
+    "(Y, \\hat{Y}) &\\mapsto f(Y, \\hat{Y})\n",
+    "\\end{align}$$\n",
+    "\n",
+    "It verifies the following properties:\n",
+    "$$\\begin{align}\n",
+    "& \\forall Y_{l} < Y_{h}: f(Y_{l}, \\hat{Y}) < f(Y_{h}, \\hat{Y}) \\\\\n",
+    "& \\forall \\hat{Y}_{l} < \\hat{Y}_{h}: f(Y, \\hat{Y}_{l}) > f(Y, \\hat{Y}_{h}) \\\\\n",
+    "& \\forall i \\neq j: R_{i,j} = f(Y_{i}, \\hat{\\mu}_{-(i,j)}(X_{i}))\n",
+    "\\end{align}$$\n",
+    "\n",
+    "## Prediction function\n",
+    "From a conformity score and a prediction, the function $g$ ouputs a perturbed prediction:\n",
+    "$$\\begin{align}\n",
+    "g:F \\times E &\\to E\\\\\n",
+    "(R, \\hat{Y}) &\\mapsto g(R, \\hat{Y})\n",
+    "\\end{align}$$\n",
+    "\n",
+    "It verifies the following properties:\n",
+    "$$\\begin{align}\n",
+    "& \\forall R_{l} < R_{h}: g(R_{l}, \\hat{Y}) < g(R_{h}, \\hat{Y}) \\\\\n",
+    "& \\forall i \\neq j: Y_{i} = g(R_{i,j}, \\hat{\\mu}_{-(i,j)}(X_{i}))\n",
+    "\\end{align}$$\n",
+    "\n",
+    "## Function relationship\n",
+    "The two functions must verify the following property:\n",
+    "$$\\begin{align}\n",
+    "\\forall \\hat{Y}, \\forall R: R = f(g(R, \\hat{Y}), \\hat{Y}) \n",
+    "\\end{align}$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88da4fd0",
+   "metadata": {},
+   "source": [
+    "___\n",
+    "# The jackknife+ prediction interval\n",
+    "Let derive the proof of the theorem to build [the jackknife+ prediction interval](https://arxiv.org/pdf/1905.02928.pdf) (Theorem 1, part 6, p. 18-21) with more general nonconformity scores.\n",
+    "\n",
+    "## R contruction\n",
```

Comment:
R-conStruction

Reply:
I proposed "Construction of R" and "Construction of A" to avoid any ambiguity.